src/methods/properties.js

```diff
@@ -1282,6 +1282,10 @@ export class PropertiesImplementation {
           enumerable: "enumerable" in X ? X.enumerable : false,
           configurable: "configurable" in X ? X.configurable : false,
         });
+      } else if (O.isPartialObject() && value instanceof Value && !(value instanceof AbstractValue)) {
+        let realmGenerator = realm.generator;
+        invariant(realmGenerator);
+        if (typeof P === "string") realmGenerator.emitFullInvariant(O, P, value);
```

Comment:
Do we want to do this if O is partial but not intrinsic? If not, then line 1278 above also needs attention.

Reply:
Good point. Will address.